contracts/BitmaskRBAC.sol

```diff
@@ -128,15 +149,16 @@ contract BitmaskRBAC is RBAC {
     checkRoleExists(_role)
     returns (bool)
   {
-    return super.hasRole(_operator, _role);
+    uint256 bitmask = users[_operator].roleBitmask;
+    uint position = roleBitIndices[_role];
+    return bitmask.hasBit(position);
   }
 
-  function _revokeRoleIfGrantedNoChecks(address user, string roleName)
-  internal {
-    if (hasRole(user, roleName)) {
-      userCountsByRole[roleName]--;
-      removeRole(user, roleName);
-    }
+  function removeRole(address _operator, string _roleName) internal {
```

Comment:
I think removeRole is dangerous for the same reason addRole was dangerous. Any subclass calling it with an unrecognised roll will remove RBAC_ADMIN privileges. Suggest we fold this in to revokeRole in the same way we did for grantRole.

Reply:
Honestly I disagree, I don't think either are dangerous.  They are internal functions that make the code easier to read and emulate the methods of the code that we were previously inheriting.

As an aside, I can imagine a use case where you would want to delete all the RBAC_ADMINs, maybe you want to have an rbac admin go in, set all the users up, and then delete themselves.

Reply:
> emulate the methods of the code that we were previously inheriting.

The code we were previously inheriting wouldn't revoke the rbac_admin role if it was accidentally called with an unrecognised role.

> As an aside, I can imagine a use case where you would want to delete all the RBAC_ADMINs

Agree.

> Nor does having one rbac admin guarantee that user roles can be functionally changed

Indeed. So we can meet the use case you mentioned above. It's just harder to do it by mistake.